data/scripts/creaturescripts/modal_window_helper.lua

```diff
@@ -0,0 +1,197 @@
+local ModalWindowAutoID = 10000
+local buttonOrder = {
+    [4] = {2, 3, 4, 1},
+    [3] = {2, 3, 1},
+    [2] = {2, 1},
+    [1] = {1}
+}
+
+if not __ModalWindow then
+    __ModalWindow = ModalWindow
+    ModalWindows = {}
+end
+
+ModalWindow = {}
+ModalWindow.__index = ModalWindow
+
+function ModalWindow.new(self, ...)
+    if type(self) ~= "table" then
+        return __ModalWindow(self, ...)
+    end
+
+    self.modalWindowId = 0
+    self.buttons = {}
+    self.choices = {}
+    self.using = 0
+    return setmetatable(self, ModalWindow)
+end
+
+setmetatable(ModalWindow, {
+    __call = function (self, ...)
+        return ModalWindow.new(...)
+    end
+})
+
+function ModalWindow:setTitle(title)
+    self.title = title
+end
+
+function ModalWindow:setMessage(message)
+    self.message = message
+end
+
+function ModalWindow:addButton(name, callback)
+    for _, button in pairs(self.buttons) do
+        if button.name == name then
+            button.callback = callback
```

Comment:
It should refuse, not silently override. Reusing the button name probably indicates a mistake.

Reply:
This suggestion can be applied with other complications, for example: how do we know the index of the button?... easy, using a local variable to count inside the loop. Ok, now let's go to the onModalWindow event, the only reference to the button is the ID, how do I find the button? any suggestion that does not have to do with creating new tables with duplicate buttons to comply with quick access in all cases?
![image](https://user-images.githubusercontent.com/28090948/194660969-1e2442cb-2f06-4b9d-99df-10cdfc68706f.png)
maybe?

otherwise we return to this point again, only this time in the part where the user interacts with the window instead of in the construction of the modalwindow.

millions and millions of windows can be created in less than 1 second, I don't understand what is the point of optimizing this?